contracts/protocol/facets/PriceDiscoveryHandlerFacet.sol

```diff
@@ -0,0 +1,143 @@
+ */
+contract PriceDiscoveryHandlerFacet is IBosonPriceDiscoveryHandler, PriceDiscoveryBase, BuyerBase {
+    /**
+     * @notice Facet Initializer
+     * This function is callable only once.
+     */
+    function initialize() public onlyUninitialized(type(IBosonPriceDiscoveryHandler).interfaceId) {
+        DiamondLib.addSupportedInterface(type(IBosonPriceDiscoveryHandler).interfaceId);
+    }
+
+    /**
+     * @notice Commits to a price discovery offer (first step of an exchange).
+     *
+     * Emits a BuyerCommitted event if successful.
+     * Issues a voucher to the buyer address.
+     *
+     * Reverts if:
+     * - Offer price type is not price discovery. See BosonTypes.PriceType
+     * - Price discovery contract address is zero
+     * - Price discovery calldata is empty
+     * - Exchange exists already
+     * - Offer has been voided
+     * - Offer has expired
+     * - Offer is not yet available for commits
+     * - Buyer address is zero
+     * - Buyer account is inactive
+     * - Buyer is token-gated (conditional commit requirements not met or already used)
+     * - Any reason that PriceDiscoveryBase fulfilOrder reverts. See PriceDiscoveryBase.fulfilOrder
+     * - Any reason that ExchangeHandler onPremintedVoucherTransfer reverts. See ExchangeHandler.onPremintedVoucherTransfer
+     *
+     * @param _buyer - the buyer's address (caller can commit on behalf of a buyer)
+     * @param _tokenIdOrOfferId - the id of the offer to commit to or the id of the voucher (if pre-minted)
+     * @param _priceDiscovery - price discovery data (if applicable). See BosonTypes.PriceDiscovery
+     */
+    function commitToPriceDiscoveryOffer(
+        address payable _buyer,
+        uint256 _tokenIdOrOfferId,
+        PriceDiscovery calldata _priceDiscovery
+    ) external payable override exchangesNotPaused buyersNotPaused {
+        // Make sure caller provided price discovery data
+        require(
+            _priceDiscovery.priceDiscoveryContract != address(0) && _priceDiscovery.priceDiscoveryData.length > 0,
+            INVALID_PRICE_DISCOVERY
+        );
+
+        bool isTokenId;
+        uint256 offerId;
+
+        // First try to fetch offer with _tokenIdOrOfferId
+        (bool exists, Offer storage offer) = fetchOffer(_tokenIdOrOfferId);
+
+        if (exists) {
+            // Set offer id if offer exists
+            offerId = _tokenIdOrOfferId;
+        } else {
+            // Extract offerId from _tokenIdOrOfferId
+            offerId = _tokenIdOrOfferId >> 128;
+
+            // Fetch offer with offerId
+            (exists, offer) = fetchOffer(offerId);
+
+            // Make sure offer exists
+            require(exists, NO_SUCH_OFFER);
+
+            // Sinalize that _tokenIdOrOfferId is a token id
+            isTokenId = true;
+        }
+
+        // Make sure offer exists, is available, and isn't void, expired, or sold out
+        require(exists, NO_SUCH_OFFER);
```

Comment:
comment doesnt match with what the require checks? if so, there is the same require in the else clause above that could be removed